src/main/resources/io/jenkins/plugins/echarts/render-configurable-trend-chart.js

```diff
@@ -3,9 +3,10 @@ window.addEventListener("DOMContentLoaded", () => {
 
     dataHolders.forEach(dataHolder => {
         const chartId = dataHolder.getAttribute("data-chart-id");
-        const enableLinks = dataHolder.getAttribute("data-enable-links") || false;
+        const enableLinks = dataHolder.getAttribute("data-enable-links");
         const configurationId = dataHolder.getAttribute("data-configuration-id");
+        const generatedId = dataHolder.getAttribute("data-generated-id");
 
-        echartsJenkinsApi.renderConfigurableTrendChart(`${chartId}`, `${enableLinks}`, `${configurationId}`, trendProxy);
+        echartsJenkinsApi.renderConfigurableTrendChart(`${chartId}`, `${enableLinks}`, `${configurationId}`, window[`trendProxy${generatedId}`]);
```

Comment:
Can you explain what the `window[...]` does? It works but I do not understand the semantics.

Reply:
It's a way of accessing a global variable.
The variable is declared by the internals of `<st:bind value="${it}" var="trendProxy${generatedId}" />` and it's in the global scope. I do not know upfront what the name of the variable will be because `generatedId` is generated in runtime. I'm passing the `generatedId` as a data attribute to the script so that I can access the object the name of which I can only know in runtime.

I could alternatively pass the name of the variable to the script instead of `generatedId` as I did in https://github.com/jenkinsci/warnings-ng-plugin/pull/1862. Though the `window[...]` syntax still stays.

If the name was static we would simply access the variable by its name (like e.g. `echartsJenkinsApi` is accessed on this exact line).

Let me know if it's more clear now.

Reply:
I see thanks! So this trick is to get the actual value while using the key `trendProxy${generatedId}`.